### ISSUE TITLE
Warning when looking for nipype submodules.

### ISSUE THREAD

**Issue by alexsavio** [duecredit/duecredit#78]
Hi,

I am getting a warning message when using duecredit with nipype.  Please check below.
Might be the way you are checking for the existence of submodules? Could you point where in the duecredit code this is?

```
2016-05-04 16:14:32,911 [WARNING] Could not find fsl in module <module 'nipype.interfaces' from '/home/alexandre/Software/nipype/nipype/interfaces/__init__.py'>: module 'nipype.interfaces' has no attribute 'fsl' (injector.py:193)
160504-16:14:32,911 duecredit WARNING:
	 Could not find fsl in module <module 'nipype.interfaces' from '/home/alexandre/Software/nipype/nipype/interfaces/__init__.py'>: module 'nipype.interfaces' has no attribute 'fsl'
2016-05-04 16:14:32,911 [WARNING] Could not find spm in module <module 'nipype.interfaces' from '/home/alexandre/Software/nipype/nipype/interfaces/__init__.py'>: module 'nipype.interfaces' has no attribute 'spm' (injector.py:193)
160504-16:14:32,911 duecredit WARNING:
	 Could not find spm in module <module 'nipype.interfaces' from '/home/alexandre/Software/nipype/nipype/interfaces/__init__.py'>: module 'nipype.interfaces' has no attribute 'spm'
```

Thanks!

**Comment by yarikoptic** [comment]
>  2016-05-04 16:14:32,911 [WARNING] Could not find fsl in module <module 'nipype.interfaces' from '/home/alexandre/Software/nipype/nipype/interfaces/__init__.py'>: module 'nipype.interfaces' has no attribute 'fsl' (injector.py:193)
that is the location -- "injector.py:193" ;)

**Comment by yarikoptic** [comment]
could you give a minimal snapshot (or just point to your branch of nipype or whatnot) on how to provoke such behavior?  the problem is that we are checking too early since `fsl` not being imported yet is not available within `interfaces`:
```
In [1]: import nipype.interfaces

In [2]: dir(nipype.interfaces)
Out[2]: 
['DataGrabber',
 'DataSink',
 'Function',
 'IdentityInterface',
 'Merge',
 'Rename',
 'Select',
 'SelectFiles',
 '__builtins__',
 '__doc__',
 '__docformat__',
 '__file__',
 '__name__',
 '__package__',
 '__path__',
 'base',
 'io',
 'traits_extension',
 'utility']

*In [3]: import nipype.interfaces.fsl

In [4]: dir(nipype.interfaces)
Out[4]: 
['DataGrabber',
 'DataSink',
 'Function',
 'IdentityInterface',
 'Merge',
 'Rename',
 'Select',
 'SelectFiles',
 '__builtins__',
 '__doc__',
 '__docformat__',
 '__file__',
 '__name__',
 '__package__',
 '__path__',
 'base',
 'fsl',
 'io',
 'traits_extension',
 'utility']
```

**Comment by alexsavio** [comment]
Ooops I see, injector.py line 193. :+1:  

I am using the nipype master branch. I will have a look too.

**Comment by alexsavio** [comment]
Another approach could be `try and __import__`, instead of `dir` ?

**Comment by yarikoptic** [comment]
well -- the point is that duecredit should nohow interfer or do anything custom (such as imports)... matteo has mentioned that you are using python 3.5 which we apparently don't travis against yet... we will rerun our tests under 3.5 (https://github.com/duecredit/duecredit/pull/79) and see it could be something specific to it

interestingly that nipype/__init__.py also doesn't import those submodules... so it would be interesting to replicate to see what causes the warnings and how dangerous it is all is or could be ignored ;-)

**Comment by yarikoptic** [comment]
at least our tests didn't detect anything 3.5 special https://travis-ci.org/duecredit/duecredit/builds/127833554

**Comment by yarikoptic** [comment]
ok -- replicated even with python2.7. we will look into it

**Comment by alexsavio** [comment]
I wonder if the nipype injector makes sense now that we will use duecredit directly from nipype.

**Comment by yarikoptic** [comment]
you are correct -- nipy injection would not be needed as soon as nipype supports it directly.  
So far I thought to support decision on either to inject or not based on the module version, but may be I could/should add additional ability to check, e.g. if `nipype.due` is available etc, so even in development snapshots could avoid unnecessary injection.
But I believe that this issue is independent to the warnings -- they poped up even without trying your PR with proper duecredit entry
